doc/whats_new.rst

```diff
@@ -203,6 +203,10 @@ Bug
 
 API
 ~~~
+- Remove :func:`mne.io.read_annotations_brainvision`, :func:`mne.io.read_annotations_eeglab`,
```

Comment:
Once you remove names from `python_reference.rst` you should use double-backticks and no `:func:` on their names here, and anywhere else they show up in `whats_new.rst`. Otherwise it leads to Sphinx build warnings, if you look at the CircleCI build log you will see:
```
/home/circleci/project/doc/whats_new.rst:27: WARNING: py:func reference target not found: mne.io.read_annotations_eeglab
/home/circleci/project/doc/whats_new.rst:91: WARNING: py:func reference target not found: mne.io.read_annotations_brainvision
/home/circleci/project/doc/whats_new.rst:105: WARNING: py:func reference target not found: mne.beamformer.make_lcmv`
/home/circleci/project/doc/whats_new.rst:206: WARNING: py:func reference target not found: mne.io.read_annotations_brainvision
/home/circleci/project/doc/whats_new.rst:206: WARNING: py:func reference target not found: mne.io.read_annotations_eeglab
/home/circleci/project/doc/whats_new.rst:207: WARNING: py:func reference target not found: mne.io.edf.edf.read_annotations_edf
/home/circleci/project/doc/whats_new.rst:207: WARNING: py:func reference target not found: mne.annotations.read_brainstorm_annotations
/home/circleci/project/doc/whats_new.rst:209: WARNING: py:func reference target not found: mne.io.read_annotations_eeglab
/home/circleci/project/doc/whats_new.rst:303: WARNING: py:func reference target not found: mne.io.read_annotations_eeglab
```